graylog2-server/src/main/java/org/graylog2/migrations/V20190705071400_AddEventIndexSetsMigration.java

```diff
@@ -84,41 +105,68 @@ public void upgrade() {
                 "Stores Graylog events.",
                 elasticsearchConfiguration.getDefaultEventsIndexPrefix(),
                 ElasticsearchConfiguration.DEFAULT_EVENTS_INDEX_PREFIX,
+                EVENT_TEMPLATE_TYPE,
                 Stream.DEFAULT_EVENTS_STREAM_ID,
                 "All events",
                 "Stream containing all events created by Graylog"
         );
+
+        deleteObsoleteStreamAndIndexSet();
         ensureEventsStreamAndIndexSet(
                 "Graylog System Events",
                 "Stores Graylog system events.",
                 elasticsearchConfiguration.getDefaultSystemEventsIndexPrefix(),
-                ElasticsearchConfiguration.DEFAULT_SYSTEM_EVENTS_INDEX_PREFIX,
-                Stream.DEFAULT_SYSTEM_EVENTS_STREAM_ID,
+                DEFAULT_SYSTEM_EVENTS_INDEX_PREFIX,
+                SYSTEM_EVENT_TEMPLATE_TYPE,
+                DEFAULT_SYSTEM_EVENTS_STREAM_ID,
                 "All system events",
                 "Stream containing all system events created by Graylog"
         );
+        ensureSystemNotificationEventsDefinition();
+    }
+
+    // Delete previously created System Events which use Event template instead of System Events template
+    private void deleteObsoleteStreamAndIndexSet() {
+        Optional<IndexSetConfig> optSystemEventConfig =
+                getEventsIndexSetConfig(elasticsearchConfiguration.getDefaultSystemEventsIndexPrefix(), EVENT_TEMPLATE_TYPE);
+        if (optSystemEventConfig.isPresent()) {
+            String id = optSystemEventConfig.get().id();
+            try {
+                for (Stream stream: streamService.loadAllWithIndexSet(id)) {
+                    streamService.destroy(stream);
+                }
+                final IndexSet indexSet = indexSetRegistry.get(id)
+                        .orElseThrow(() -> new javax.ws.rs.NotFoundException("Index set <" + id + "> not found."));
```

Comment:
```suggestion
                        .orElseThrow(() -> new IllegalStateException("Index set <" + id + "> not found."));
```
I wouldn't use that exception here. It's typically only used in Resources.